core/src/jsMain/kotlin/Peripheral.kt

```diff
@@ -54,8 +57,36 @@ public class JsPeripheral internal constructor(
     public override val services: List<DiscoveredService>?
         get() = platformServices?.map { it.toDiscoveredService() }
 
-    public override suspend fun rssi(): Int {
-        TODO("Not yet implemented")
+    private val supportsAdvertisements = js("BluetoothDevice.prototype.watchAdvertisements") != null
+
+    public override suspend fun rssi(): Int = suspendCancellableCoroutine { continuation ->
+        check(supportsAdvertisements) { "watchAdvertisements unavailable" }
+
+        var listener: ((JsEvent) -> Unit)? = null
```

Comment:
If we want some null safety baked into the transpiled code (or at least on the JVM it compiles to bytecode that has `null` checks at usage sites), then you could change it to:

```suggestion
        lateinit var listener: (JsEvent) -> Unit
```

We're basically promising the compiler that we'll set the variable before we try to use it. On the JVM at least (and I assume on JS) it adds non-`null` checks at usage sites and throws `NullPointerException` if `null` when used.